gossipsub.go

```diff
@@ -323,16 +333,56 @@ func (gs *GossipSubRouter) pxConnect(peers []*pb.PeerInfo) {
 		return
 	}
 
-	// initiate connections, without blocking the event loop
-	go func() {
+	// schedule connections
+	gs.connectMx.Lock()
+	if len(gs.connectPend) < GossipSubMaxPendingConnections {
 		for _, ci := range toconnect {
+			// don't drop a signed routing state if we already have one and the px record is missing
+			_, ok := gs.connectPend[ci.p]
+			if !ok || ci.srr != nil {
+				gs.connectPend[ci.p] = ci.srr
+			}
+		}
+		select {
+		case gs.connectSched <- struct{}{}:
+		default:
+		}
+	} else {
+		log.Debug("ignoring peers obtained through px; too many pending connections")
+	}
+	gs.connectMx.Unlock()
+}
+
+func (gs *GossipSubRouter) connectScheduler() {
+	for {
+		gs.connectMx.Lock()
+		var ci connectInfo
+		var pend bool
```

Comment:
`found`? `pend` doesn't mean anything to me (I know what you're saying, but it's still confusing).